app/src/main/java/com/example/newsportal/data/local/LocalDataSource.kt

```diff
@@ -6,18 +6,39 @@ import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
 class LocalDataSource(
-    private val dao: NewsDao,
+    private val newsDao: NewsDao,
+    private val bookmarksDao: BookmarksDao,
     private val mapper: ArticleDatabaseMapper
 ) : ILocalDataSource {
 
     override fun getAllNews(): Flow<ResultWrapper<List<Article>>> {
-        return dao.getAll().map { databaseList ->
+        return newsDao.getAll().map { databaseList ->
             ResultWrapper.Success(databaseList.map (mapper::mapToDomain) )
         }
     }
 
     override fun insertNews(news: List<Article>) {
         val mapped = news.map { mapper.mapToLocal(it) }
-        dao.add(mapped)
+        newsDao.add(mapped)
+    }
+
+    override fun addBookmark(article: Article) {
+        val local = mapper.mapToLocal(article)

```

Comment:
If you use variableone time, local variable will be useless. Suggest to use `newsDao.updateArticle(mapper.mapToLocal(article))` and  `bookmarksDao.deleteBookmark(mapper.mapDomainToBookmark(article))`